fea-rs/src/compile/lookups.rs

```diff
@@ -975,13 +975,24 @@ impl SomeLookup {
         }
     }
 
-    pub(crate) fn add_gsub_type_4(&mut self, target: Vec<GlyphId16>, replacement: GlyphId16) {
+    /// Returns `true` if this replacement shadows an existing rule.
+    ///
+    /// In this case the rule is not added, and the client should report an error.
```

Comment:
> In this case the rule is not added...

I'd expect then for a method named `add_gsub_type_4` to logically return `false` in this situation, rather than `true`, no?

Reply:
or is it like with unix commands, zero (false) means success, non-zero means some error?

Reply:
for me it's like the `insert` method on rust's set types; it returns `true` if the item existed.

This isn't my favourite API in the world but it gets the job done. 🤷